crates/ruff_linter/src/rules/pyflakes/rules/unused_import.rs

```diff
@@ -138,9 +158,31 @@ fn is_first_party(qualified_name: &str, level: u32, checker: &Checker) -> bool {
     }
 }
 
+/// Find the `ExprList` for a top level `__all__`, if one exists.
+fn find_dunder_all_expr<'a>(semantic: &'a SemanticModel) -> Option<&'a ast::ExprList> {
+    let stmt = semantic
+        .global_scope()
+        .get_all("__all__")
+        .map(|binding_id| &semantic.bindings[binding_id])
+        .find_map(|binding| match binding.kind {
+            BindingKind::Export(_) => binding.statement(semantic),
+            _ => None,
+        })?;
+    let expr = match stmt {
+        Stmt::Assign(ast::StmtAssign { value, .. }) => Some(&**value),
+        Stmt::AnnAssign(ast::StmtAnnAssign { value, .. }) => value.as_deref(),
+        Stmt::AugAssign(ast::StmtAugAssign { value, .. }) => Some(&**value),
+        _ => None,
+    }?;
+    match expr {
+        ast::Expr::List(list) => Some(list),
```

Comment:
I think it's also valid to use a set here (CC: @AlexWaygood)

Reply:
ah, it was a tuple! Thanks

Reply:
Oops, I found that there is already a data structure for this in `ruff_python_ast`. Will fix.

Reply:
Ended up using `Expr`